drgn_tools/iscsi.py

```diff
@@ -73,76 +241,15 @@ def print_iscsi_sessions(prog: Program) -> None:
         print(msg)
         return
 
-    output = {}
-
     for session in reversed(list(for_each_iscsi_session(prog))):
-        print("**********")
         conn = session.leadconn
 
-        persistent_address = escape_ascii_string(
-            conn.persistent_address.string_()
-        )
-        persistent_port = int(conn.persistent_port)
-        output[
-            "Scsi_Host"
-        ] = f"host{session.host.host_no.value_()} ({hex(session.host.value_())})"
-        output["Session"] = hex(session.address_of_())
-        output["SID"] = str(int(session.cls_session.sid))
-        output["Persistent Portal"] = f"{persistent_address}:{persistent_port}"
-        output["Iface Name"] = escape_ascii_string(session.ifacename.string_())
-        output["Session State"] = ISCSI_SESSION_STATES[
-            session.cls_session.state
-        ]
-        connstate = enum_type_to_class(
-            prog.type("enum iscsi_connection_state"), "connstate"
-        )
-        output["Connection State"] = str(
-            enum_name_get(
-                connstate,
-                conn.cls_conn.state,
-                "UNKNOWN",
-            )
-        )
-        output["Initiatorname"] = escape_ascii_string(
-            session.initiatorname.string_()
-        )
-        output["Targetname"] = escape_ascii_string(
-            session.targetname.string_()
-        )
-
-        print_dictionary(output)
-
-        print("Attached SCSI devices: ")
-        print("**********")
-        print(
-            "Host Number: {}  STATE: {}".format(
-                session.host.host_no.value_(),
-                session.host.shost_state.format_(type_name=False),
-            )
-        )
+        iscsi_info = get_iscsi_info(conn, prog)
+        print_iscsi_info(iscsi_info)
 
         for scsi_dev in for_each_scsi_host_device(prog, session.host):
-            name = scsi_device_name(prog, scsi_dev)
-            print(
-                "scsi{} Channel {} Id {} Lun: {}".format(
-                    session.host.host_no.value_(),
-                    int(scsi_dev.channel),
-                    int(scsi_dev.id),
-                    int(scsi_dev.lun),
-                )
-            )
-            sdev_state = enum_type_to_class(
-                prog.type("enum scsi_device_state"), "sdev_state"
-            )
-            state = enum_name_get(
-                sdev_state,
-                scsi_dev.sdev_state,
-                "UNKNOWN",
-            )
-            print("  Attached iscsi disk: {}  State: {}".format(name, state))
-
-        print()
-
+            iscsi_disks_info = get_iscsi_disks_info(scsi_dev, prog)
+            print_iscsi_info(iscsi_disks_info)
```

Comment:
Could this be a table instead? It would probably make it easier to read. See the `drgn_tools/table.py`.